File: hw/chicaiza/u3/hw26Guis/SchoolSystemJs/ec/edu/espe/schoolsystem/utils/MongoConnection.js
// ec/edu/espe/schoolsystem/utils/MongoConnection.js
const mongoose = require('mongoose');

class MongoConnection {
    static async connect() {
        try {
            await mongoose.connect(
                'mongodb+srv://daniel:daniel2007@cluster0.v7buh9x.mongodb.net/school?appName=Cluster0'
            );
            console.log('MongoDB connected');
        } catch (error) {
            console.error('MongoDB connection error:', error.message);
        }
    }
}

module.exports = MongoConnection;

File: hw/hidalgo/u3/hw28PrinciplesAplication/java/ToolsManagement/src/main/java/ec/edu/espe/tools/utils/MongoConnection.java
package ec.edu.espe.tools.utils;

import com.mongodb.client.MongoClient;
import com.mongodb.client.MongoClients;
import com.mongodb.client.MongoDatabase;

/**
 *
 * @author Mikaeel Hidalgo, Paradigm, @ESPE
 */
public class MongoConnection {

    private static MongoConnection instance;
    private MongoClient mongoClient;
    private MongoDatabase database;
    private static final String CONNECTION_STRING = "mongodb+srv://Mikael:<Mikael1897>@cluster0.fpyoe9m.mongodb.net/?appName=Cluster0";
    private static final String DATABASE_NAME = "ToolsDB";

    private MongoConnection() {
        try {
            mongoClient = MongoClients.create(CONNECTION_STRING);
            database = mongoClient.getDatabase(DATABASE_NAME);
            System.out.println("Sucessful connection");
        } catch (Exception e) {
            System.err.println("Error in connection " + e.getMessage());
        }
    }

    public static MongoConnection getInstance() {
        if (instance == null) {
            instance = new MongoConnection();
        }
        return instance;
    }

    public MongoDatabase getDatabase() {
        return database;
    }
}
